lib/active_record_shards/default_replica_patches.rb

```diff
@@ -119,7 +119,11 @@ def on_replica_unless_tx
     alias_method :on_slave_unless_tx, :on_replica_unless_tx
 
     def force_on_replica(&block)
```

Comment:
I think there’s a couple of problem with this code, that makes this change hard to review.

1. `#force_on_replica` isn’t a great name for this method. What it does it that it calls some method on the replica database _even_ if the model doesn’t say `on_replica_by_default`. I think mainly the schema loading methods are meant to use this method, and not the data loading methods.
2. #on_replica_unless_tx` and `force_on_replica` call different methods (`on_replica` and `on_cx_switch_block`) that take different arguments. This is because `force_on_replica` needs to pass e.g. the `:force` argument, which cannot be passed to `on_replica`.